pkg/util/iptables/iptables.go

```diff
@@ -284,6 +284,42 @@ func (runner *runner) restoreInternal(args []string, data []byte, flush FlushFla
 	return nil
 }
 
+// ParseTableSaveChains parses a table's save and returns a map of Chain to chain in iptables-save format
+// (with counters etc).
+func ParseTableSaveChains(table Table, save []byte) map[Chain]string {
+	// get lines
+	lines := strings.Split(string(save), "\n")
+	chainsMap := make(map[Chain]string)
+	tablePrefix := "*" + string(table)
+	lineNum := 0
+	// find beginning of table
+	for ; lineNum < len(lines); lineNum++ {
+		line := lines[lineNum]
```

Comment:
this makes a copy of every line.  It's minor, but let's not do that for such a small loop